### DISCUSSION TITLE
Passing mod, value arguments from component on mouseDown to another function

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#264] [Q&A] [answered]
Hello,
Instead of enjoying the nice weather, I stayed home to work on my project. It seems that I'm a little bit nerdy :)

I have some issue related to modulator/component when the mouse is down on it.

Ideally I would go for` function SendSysexOnModDown(mod, value, source)`
But since it's for a mouseDown action it has to be component not modulator related :   `function SendSysexOnModDown(comp, event)`

In my **SendSysexOnModDown** function I need to pass the modulator clicked on, its value to another function.
I tried getting the name of the modulator then passing it to the other function. 

```
function SendSysexOnModDown(comp, event)

ModName = L(comp:getOwner():getName())
CCParamAssign(ModName) -- pass the modulator name as argument
...

end
```

And the sub-function called when clicked on :

```
function CCParamAssign(mod, value, source) 

ParamPos = mod:getComponent():getProperty("componentGroupName")

if ParamPos == "PreFxGrp" then
	ParamCtrl1PosValue = 0
elseif ParamPos == "MainFxGrp" then
	ParamCtrl1PosValue = 1
elseif ParamPos == "PostFxGrp" then
	ParamCtrl1PosValue = 2
end

...
end -- end function
```

It's not working of course because I need to start my 2nd function with :
`function CCParamAssign(ModName) `

My question is, how to say the following, knowing only the name of the modulator instead of "mod" if it was a "LuaModulatorValueChange" function  : 
ParamPos = **mod**:getComponent():getProperty("componentGroupName")

I tried the following but it doesn't work :
`mod = panel:getModulatorByName(ModName):getComponent()`

I just need to call the modulator of the component clicked on, which varies (I have about 100 mod on my pannel).
I struggle with this mousedown event since it's based on component and not modulator. It is probably very easy but as usual when it's too obvious it's problematic for me :)

Thanks for your help. 

Damien

**Comment by damiensellier** [comment, marked as answer]
OK Solved I found the proper syntax
`
ParamPos = panel:getModulatorByName(ModName):getComponent():getProperty("componentGroupName")`

so the function called related to the modulator with mouseDown is : 

```
function CCParamAssign(ModName) 

ParamPos = panel:getModulatorByName(ModName):getComponent():getProperty("componentGroupName")`

if ParamPos == "PreFxGrp" then
	ParamCtrl1PosValue = 0
elseif ParamPos == "MainFxGrp" then
	ParamCtrl1PosValue = 1
elseif ParamPos == "PostFxGrp" then
	ParamCtrl1PosValue = 2
end

...
end -- end function
```

Have a nice weekend

**Comment by dobo365** [comment]
Reading your post in my mail I was going to tell you that you were missing the "value" and "source" arguments of your call to your function CCParamAssign.

You call it with just "mod"
```
ModName = L(comp:getOwner():getName())
CCParamAssign(ModName) -- pass the modulator name as argument
```
while you define it with 3 arguments 
`function CCParamAssign(mod, value, source)`

But I now see you changed that. Good.
